altair/api.py

```diff
@@ -409,6 +409,70 @@ def to_dict(self):
         D['config'] = self.vlconfig.to_dict()
         return D
 
+    def select_x(self):
+        """
+        Helper function that does a best effort of selecting an automatic x axis.
+
+        Raises AssertionError if it cannot find a x axis to choose.
+        """
+        assert self.data is not None
+        assert len(self.data) >= 1
+
+        d = self._classify_data_by_type()
+
+        # Choose the first column found on the following order: T, O, N, Q
+        chosen_x = None
+        for typ in ['Q', 'N', 'O', 'T']:
+            if len(d[typ]) >= 1:
+                chosen_x = d[typ][0]
+
+        if chosen_x is None:
+            raise AssertionError('Could not select X.')
```

Comment:
Also, I do prefer ValueError. I think assertion errors should be reserved
for explicit assertions in Python.

On Wed, Oct 21, 2015 at 10:27 PM, Alejandro Guerrero Gonzalez <
notifications@github.com> wrote:

> In altair/api.py
> <https://github.com/ellisonbg/altair/pull/78#discussion_r42705906>:
>
> > +
> > +        Raises AssertionError if it cannot find a x axis to choose.
> > +        """
> > +        assert self.data is not None
> > +        assert len(self.data) >= 1
> > +
> > +        d = self._classify_data_by_type()
> > +
> > +        # Choose the first column found on the following order: T, O, N, Q
> > +        chosen_x = None
> > +        for typ in ['Q', 'N', 'O', 'T']:
> > +            if len(d[typ]) >= 1:
> > +                chosen_x = d[typ][0]
> > +
> > +        if chosen_x is None:
> > +            raise AssertionError('Could not select X.')
>
> I completely agree @ellisonbg <https://github.com/ellisonbg>. With
> complex datasets, there's no hope of getting it right, so I'm not making
> the functions very complex.
>
> Instead, I think we should focus on creating a UI a la Polestar that will
> give users control over their graph. For this UI, we could choose to
> display a default graph as a starting point, for example.
>
> Besides the UI, these helper methods will allow us to create something
> similar to what other notebooks currently have. Users are very vocal about
> the ability to instantly visualize something, even if it's not the right
> graph at first. They later ask for tweaking, but it's the instant
> visualization of their data that they love.
>
> I think that the UI and these methods are a killer combination. These
> methods are small and simple enough that they should be easy to maintain.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ellisonbg/altair/pull/78/files#r42705906>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com